rewriter/ChainedSig.cc

```diff
@@ -0,0 +1,213 @@
+#include "rewriter/ChainedSig.h"
+#include "ast/Helpers.h"
+#include "ast/treemap/treemap.h"
+#include "core/errors/rewriter.h"
+
+using namespace std;
+namespace sorbet::rewriter {
+
+struct ChainedSigWalk {
+    vector<ast::Send *> incompleteSigs;
```

Comment:
I explored this and indeed the implementation is much easier using `preTransformSend`. I still needed an auxiliary variable, to prevent adding incorrect errors to `sig` without a block, but I don't think it's a big deal.

Let me know what you think!